src/screens/Assistant/SelectFavouritesBottomSheet.tsx

```diff
@@ -0,0 +1,249 @@
+import SectionSeparator from '@atb/components/sections/section-separator';
+import MessageBox from '@atb/components/message-box';
+
+type SelectableFavouriteDepartureData = {
+  handleSwitchFlip: (favouriteId: any) => void;
+  favouriteId: string;
+  active: boolean;
+  lineTransportationMode: AnyMode;
+  lineTransportationSubmode?: TransportSubmode;
+  lineIdentifier: string;
+  lineName: string | TranslatedString;
+  departureStation: string;
+};
+
+const SelectableFavouriteDeparture = ({
+  handleSwitchFlip,
+  favouriteId,
+  active,
+  lineTransportationMode,
+  lineTransportationSubmode,
+  lineIdentifier,
+  lineName,
+  departureStation,
+}: SelectableFavouriteDepartureData) => {
+  const styles = useStyles();
+  const {t} = useTranslation();
+
+  return (
+    <View style={styles.selectableDeparture}>
+      <View style={styles.lineModeIcon}>
+        <TransportationIcon
+          mode={lineTransportationMode}
+          subMode={lineTransportationSubmode}
+        />
+      </View>
+      <View style={styles.selectableDepartureTextView}>
+        <ThemeText type="body__primary" style={styles.lineIdentiferText}>
+          {lineIdentifier} {lineName}
+        </ThemeText>
+
+        <ThemeText type="body__secondary" style={styles.secondaryText}>
+          {t(SelectFavouriteDeparturesText.departures.from)} {departureStation}
+        </ThemeText>
+      </View>
+
+      <View>
+        <FixedSwitch
+          accessibilityHint={t(SelectFavouriteDeparturesText.switch.a11yhint)}
+          value={active}
+          onChange={() => handleSwitchFlip(favouriteId)}
+          style={[
+            styles.toggle,
+            Platform.OS === 'android' ? styles.androidToggle : styles.iosToggle,
+          ]}
+        />
+      </View>
+    </View>
+  );
+};
+
+type SelectFavouritesBottomSheetProps = {
+  close: () => void;
+};
+
+const SelectFavouritesBottomSheet = ({
+  close,
+}: SelectFavouritesBottomSheetProps) => {
+  const styles = useStyles();
+  const {t} = useTranslation();
+  const {
+    favoriteDepartures,
+    frontPageFavouriteDepartures,
+    addFrontPageFavouriteDeparture,
+    removeFrontPageFavouriteDeparture,
+  } = useFavorites();
+  const favouriteItems = favoriteDepartures ?? [];
+  const frontpageFavouriteItems = frontPageFavouriteDepartures ?? [];
+
+  const handleSwitchFlip = (favouriteId: string) => {
+    const favouriteInFavourites = favouriteItems.find(
+      (item) => item.id === favouriteId,
+    );
+
+    const favouriteInFrontpageFavourites = frontpageFavouriteItems.find(
+      (item) => item.id === favouriteId,
+    );
+
+    if (!favouriteInFrontpageFavourites) {
+      const favourite = favoriteDepartures.find(
+        (departure) => departure.id === favouriteId,
+      );
+      if (!favourite) return;
+      addFrontPageFavouriteDeparture(favourite);
+    } else if (favouriteInFavourites) {
+      removeFrontPageFavouriteDeparture(favouriteInFrontpageFavourites);
+    }
```

Comment:
Having a hard time understanding what's going on here. Tried to formulate some questions, but it may be good to try to do a refactor to make it more readable.

- Should `favouriteInFavourites` be a boolean? (You can use `.some()` instead of `.find()` if so. Also, is it strictly necessary to check if the favorite exists before you (try to) remove it? Could this work just as well?
```ts
else {
  removeFrontPageFavouriteDeparture(favouriteInFrontpageFavourites);
}
```

- Also, what's the difference between `favouriteItems` and `favoriteDepartures`? And why do you need to search for the input departure id in both?

Reply:
I flipped the whole function, so I think it should be more readable now. Any opinions, @rosvik?

```
  const handleSwitchFlip = (favouriteId: string) => {
  // Checks if favourite is currently selected as a front page favourite
    const favouriteInFrontpageFavourites = frontpageFavouriteItems.find(
      (item) => item.id === favouriteId,
    );

    // if its already in the frontpage list, it should be removed
    if (favouriteInFrontpageFavourites) {
      removeFrontPageFavouriteDeparture(favouriteInFrontpageFavourites);
    } else {
    // if its not part of the frontpage list, it should probably be added
      const favourite = favoriteDepartures.find(
        (departure) => departure.id === favouriteId,
      );
     // there is also the crazy chance of the departure not being part of the ordinary favourite list anymore, 
     // so we will need to check if the favourite has been found, we would also get a type error if we don't
      if (favourite) addFrontPageFavouriteDeparture(favourite);
    }
  };
```

Reply:
looks like you are using both "Favorite" and "Favourite" as spelling. maybe go for just one, and maybe use the  more international "Favourite"

Reply:
I like what you did in the branch now 💯